stats.py

```diff
@@ -78,26 +90,84 @@ def passive_skill_tree() -> tuple[dict, dict]:
+	(r'(.*) has (\d+)% increased Aura Effect', 'specific_aura_effect'),
+	(r'(.\d+) to Level of all (.*) Gems', 'global_level'),
+	(r'(.\d+)% to Quality of all (.*) Gems', 'global_quality'),
+	(r'Allocates (.*)', 'additional_notable')
 ]]
 
 def _parse_item(stats: Stats, item: dict):
-	for modlist in ['implicitMods', 'explicitMods', 'craftedMods']:
+	for modlist in ['implicitMods', 'explicitMods', 'craftedMods', 'enchantMods']:
 		if modlist not in item:
 			continue
 		_parse_mods(stats, item[modlist])
 
 def _parse_mods(stats: Stats, mods: list) -> None:
 	for mod in mods:
 		for regex, attr in matchers:
-			m = regex.match(mod)
+			m = regex.search(mod)
 			if m:
+				if attr == "specific_aura_effect":
+					aura_name = m.group(1)
+					if aura_name in stats.specific_aura_effect:
+						stats.specific_aura_effect[aura_name] += int(m.group(2))
+					else:
+						stats.specific_aura_effect[aura_name] = int(m.group(2))
+					continue
+
+				if attr == "global_level":
+					stats.global_gem_level_increase += parse_gem_descriptor(m.group(2).lower(), int(m.group(1)))
+					continue
+
+				if attr == "global_quality":
+					stats.global_gem_quality_increase += parse_gem_descriptor(m.group(2).lower(), int(m.group(1)))
+					continue
+
+				if attr == "additional_notable":
+					notable = m.group(1)
+					if " if you have the matching modifier on" in notable:
+						notable = notable.split(" if you have the matching modifier on")[0]
+					stats.additional_notables |= {hash_for_notable(notable)}
+					continue
+
 				value = int(m.group(1))
 				setattr(stats, attr, getattr(stats, attr) + value)
+
+def parse_gem_descriptor(descriptor: str, value: int) -> list[tuple[list, int]]:
+	if descriptor == "":
+		# since active skills and supports are mutually exclusive, we can increase both if no conditions are specified
+		return [(["active_skill"], value), (["support"], value)]
+	if "non-" in descriptor:  # handles vaal caress - decreases all gem levels and then sets all vaal gems back to 0
+		return [(["active_skill"], value), (["support"], value), ([descriptor[4:]], -value)]
+
+	conditions = []
+	if "skill" in descriptor:
+		conditions.append("active_skill")
+	if "aoe" in descriptor:
+		conditions.append("area")
+	# this list of possible tags could be fetched directly from gems.json
+	for tag in ['mark', 'strength', 'duration', 'link', 'critical', 'chaos', 'nova', 'spell', 'trigger', 'bow',
+				'attack', 'slam', 'warcry', 'guard', 'channelling', 'travel', 'strike', 'blessing', 'low_max_level',
+				'intelligence', 'cold', 'totem', 'projectile', 'orb', 'stance', 'brand', 'dexterity', 'physical',
+				'lightning', 'fire', 'aura', 'melee', 'chaining', 'herald', 'mine', 'exceptional', 'minion', 'curse', 
+				'hex', 'movement', 'vaal', 'support', 'banner', 'golem', 'trap', 'blink', 'random_element', 'arcane']:
+		if tag in descriptor:
+			conditions.append(tag)
+	return [(conditions, value)]
```

Comment:
are you always returning a list of 1 tuple so you can append it to the lists above?
probably want to just return the tuple and do `level_mods.append(parse_gem_descriptor(...))` instead

Reply:
not always. in the case where something applies to all gems, since there is no "all gems" tag we increase both active skill gems and support gems separately and thus return two tuples

Reply:
ohh right